Doc/library/stdtypes.rst

```diff
@@ -4822,35 +4822,66 @@ Generic Alias Type
-:term:`type annotations <annotation>`.
-
-Usually, the :ref:`subscription <subscriptions>` of container objects calls the
-method :meth:`__getitem__` of the object.  However, the subscription of some
-containers' classes may call the classmethod :meth:`__class_getitem__` of the
-class instead. The classmethod :meth:`__class_getitem__` should return a
-``GenericAlias`` object.
+``GenericAlias`` objects are generally created by
+:ref:`subscripting<subscriptions>` a class. They are most often used with
+container classes (e.g. :class:`list` or :class:`dict`). They are intended
+primarily for use with :term:`type annotations <annotation>`. For example,
+``list[int]`` is a ``GenericAlias`` object created by subscripting the
+:class:`list` class with the argument :class:`int`.
+
+Usually, the :ref:`subscription <subscriptions>` of an object in Python will
+call the :meth:`__getitem__<object.__getitem__>` instance method defined on the
+object's class. For example, if we have a list
+``food = ['spam', 'eggs', 'bacon']``, calling ``food[0]`` will return the same
+value as calling ``type(food).__getitem__(food, 0)``. However, if a class
+defines the classmethod :meth:`__class_getitem__<object.__class_getitem__>`,
+then the subscription of that class may instead call the class's implementation
+of ``__class_getitem__``. This should return a ``GenericAlias`` object if it is
+properly defined. For example, because the :class:`list` class defines
+``__class_getitem__``, calling ``list[str]`` is equivalent to calling
+``list.__class_getitem__(str)``, rather than
+``type(list).__getitem__(list, str)``.
 
 .. note::
-   If the :meth:`__getitem__` of the class' metaclass is present, it will take
-   precedence over the :meth:`__class_getitem__` defined in the class (see
-   :pep:`560` for more details).
-
-The ``GenericAlias`` object acts as a proxy for :term:`generic types
-<generic type>`, implementing *parameterized generics* - a specific instance
-of a generic which provides the types for container elements.
-
-The user-exposed type for the ``GenericAlias`` object can be accessed from
-:class:`types.GenericAlias` and used for :func:`isinstance` checks.  It can
-also be used to create ``GenericAlias`` objects directly.
+   If :meth:`__getitem__<object.__getitem__>` is defined by a class's
+   :term:`metaclass`, it will take precedence over a
+   :meth:`__class_getitem__<object.__class_getitem__>` classmethod defined by
+   the class. See :pep:`560` for more details.
+
+A ``GenericAlias`` object acts as a proxy for a :term:`generic type`,
+implementing *parameterized generics*.
+
+For a container class which implements
+:meth:`__class_getitem__<object.__class_getitem__>`, the argument(s) supplied
+to a :ref:`subscription<subscriptions>` of the class may indicate the type(s)
+of the elements an object contains. For example, ``list[int]`` can be used in
+type annotations to signify a :class:`list` in which all the elements are of
+type :class:`int`.
```

Comment:
This feels slightly naggy. I wonder if there's a way to not repeat the ``list[int]`` example three times for separate reasons.